third_party/move/move-vm/runtime/src/loader/modules.rs

```diff
@@ -320,11 +323,6 @@ impl Module {
                 let module_handle = module.module_handle_at(struct_handle.module);
                 let module_id = module.module_id_for_handle(module_handle);
 
-                if module_handle != module.self_handle() {
```

Comment:
Is this (and in scripts) what allows to delete struct_ty_storage? But what about the compatibility check?

Reply:
https://github.com/aptos-labs/aptos-core/blob/8825060eb3db7f610616d3cb7f5b54b6e53f930e/third_party/move/move-bytecode-verifier/src/dependencies.rs#L237

Reply:
as long as it's definitely checked during loading (once you separate things out let's have mock tests for all those checks)